src/priority_fee.rs

```diff
@@ -320,6 +327,85 @@ impl PriorityFeeTracker {
         );
         micro_lamport_priority_fee_estimates
     }
+
+    /*
+    Algo2: given the list of accounts the algorithm will:
+     1. collect all the transactions fees over n slots
+     2. for each specified account collect the fees and calculate the percentiles
+     4. choose maximum values for each percentile between all transactions and each account
+     */
+    pub fn get_priority_fee_estimates2(
+        &self,
+        accounts: Vec<Pubkey>,
+        include_vote: bool,
+        lookback_period: Option<u32>,
+    ) -> MicroLamportPriorityFeeEstimates {
+        let start = std::time::Instant::now();
+        let lookback = calculate_lookback_size(&lookback_period, self.slot_cache.len());
+
+        let mut slots_vec = Vec::with_capacity(lookback);
```

Comment:
I noticed that in other algo we were iterating over values from the map. If I understand correctly the map impl in Rust they do not guarantee ordering which means we might be iterating the slots in random order instead of newest slot to latest slot. Instead I am using the data from slot cache and populating the vector with slots, sorting and iterating over them in reverse order...
Also I chose to iterate over slots multiple times and compare each account's data to previously calculated max from previous account... instead of iterating once but also using more memory, for each account fee vector.

Reply:
Thats a great point, there is also a crate we've used elsewhere called `indexmap` that you could switch the map to use

Reply:
I'll keep this in mind as it would probably be a better approach cause otherwise we have to allocate that vector every time method called.